### PR TITLE
refactor[next]: itir embedded: cleaner closure run

### PR DESCRIPTION
**Pull request by havogt**
This enables to run the closure also outside of the fendef context, e.g. from field view embedded.

### CODE REVIEW

File: src/gt4py/next/iterator/embedded.py [hunk 1]
```diff
@@ -1508,42 +1501,43 @@ def _validate_domain(domain: Domain, offset_provider: OffsetProvider) -> None:
...
-
-        column_range = None
-        column: Optional[ColumnDescriptor] = None
-        if kwargs.get("column_axis") and kwargs["column_axis"].value in domain:
-            column_axis = kwargs["column_axis"]
+@runtime.closure.register(EMBEDDED)
+def closure(
+    domain_: Domain,
+    sten: Callable[..., Any],
+    out,  #: MutableLocatedField,
```
Comment (havogt) [inline src/gt4py/next/iterator/embedded.py:1508]: I briefly looked into that commented type hint, I don't want to spend time resolving it...

**Review by egparedes — COMMENTED** [review]
Looks good. My only concern is about improving the documentation and usage of the `new_context()` function.

File: src/gt4py/next/iterator/embedded.py [hunk 2]
```diff
@@ -1508,42 +1501,43 @@ def _validate_domain(domain: Domain, offset_provider: OffsetProvider) -> None:
...
+    ins: list[common.Field],
+) -> None:
+    offset_provider = embedded_context.offset_provider.get()
+    _validate_domain(domain_, offset_provider)
+    domain: dict[Tag, range] = _dimension_to_tag(domain_)
+    if not (isinstance(out, common.Field) or is_tuple_of_field(out)):
+        raise TypeError("'Out' needs to be a located field.")
+
+    new_context: dict[str, Any] = {"offset_provider": offset_provider}
```
[the remark this answers is not in the document]
Comment (havogt) [reply src/gt4py/next/iterator/embedded.py:1517]: done

File: src/gt4py/next/embedded/context.py [hunk 3]
```diff
@@ -62,4 +62,6 @@ def ctx_updater(*args: tuple[cvars.ContextVar[Any], Any]) -> None:
 
 
 def within_context() -> bool:
-    return offset_provider.get() is not _undefined_offset_provider
+    return (
+        offset_provider.get() is not _undefined_offset_provider
+    )  # TODO(havogt): this is broken: if there are no shifts, the offset_provider can be empty even within context
```
[the remark this answers is not in the document]
Comment (havogt) [reply src/gt4py/next/embedded/context.py:67]: I don't understand why you want to set a default inside new_context().

File: src/gt4py/next/embedded/context.py [hunk 4]
```diff
@@ -62,4 +60,4 @@ def ctx_updater(*args: tuple[cvars.ContextVar[Any], Any]) -> None:
 
 
 def within_context() -> bool:
```
Comment (egparedes) [inline src/gt4py/next/embedded/context.py:62]: I'd rename it to something like `within_valid_context()`  and add a docstring to this function also.

File: src/gt4py/next/iterator/embedded.py (hunk 1 above)
Comment (egparedes) [reply src/gt4py/next/iterator/embedded.py:1503]: It makes sense.

**Review by egparedes — CHANGES_REQUESTED** [review]

**Review by egparedes — APPROVED** [review]
ILGTM.

**Pull request merged**